packages/scratch-gui/src/lib/legacy-storage.ts

```diff
@@ -69,6 +69,9 @@ export class LegacyStorage implements GUIStorage {
         vmState: string,
         params: { originalId: string; isCopy: boolean; isRemix: boolean; title: string; }
     ): Promise<{ id: string | number; }> {
+        if (!this.projectHost) {
+            return Promise.reject(new Error('Project host not set'));
+        }
```

Comment:
Without this, TS said that passing `this.projectHost` to `saveProjectToServer` was invalid because it might be `undefined`.